### ISSUE TITLE
Is C++17  required in 0.12.0?

### ISSUE THREAD

**Issue by toge** [jupyter-xeus/xproperty#69]
In README.md, xproperty requires C++14:

> xproperty requires a modern C++ compiler supporting C++14.

But xproperty/0.12.0 seems to require C++17 because std::any is used.
https://github.com/jupyter-xeus/xproperty/blob/0.12.0/include/xproperty/xobserved.hpp#L99

If you continue to support C++14, I believe the code needs to be modified.
If C++17 is now required, the README.md needs to be modified.

Could you please tell me what the project policy is?

**Comment by perseoGI** [comment]
Ping from `Conan`

**Comment by JohanMabille** [comment]
Indeed, xproperty 0.12 and above now require a compiler supporting C++17.
